File: frontend/src/types/commissionLedger.ts
export type CommissionStage =
  | "EXPECTED"
  | "SUBMITTED"
  | "APPROVED"
  | "PAYABLE"
  | "PAID"
  | "DISPUTED";

export interface CommissionItem {
  id: number;
  organization_id?: number | null;
  booking_id: number;
  booking_number: string;
  unit_number: string;
  lead_name: string;
  builder_name: string;
  project_name: string;
  executive_name: string;
  broker_name?: string | null;
  total_deal_value: number;

  builder_commission_rate: number;
  builder_commission_amount: number;
  gst_rate: number;
  gst_amount: number;
  tds_rate: number;
  tds_amount: number;
  net_receivable: number;

  executive_commission_rate: number;
  executive_commission_amount: number;
  company_margin_amount: number;

  stage: string;
  payout_status: string;
  invoice_number?: string | null;
  invoice_date?: string | null;
  due_date?: string | null;
  paid_date?: string | null;
  payment_reference?: string | null;

  aging_bucket: string;
  days_overdue: number;
  remarks?: string | null;
  created_at: string;
}

export interface CommissionAgingBucketSummary {
  bucket: string;
  count: number;
  total_amount: number;
}

export interface CommissionCommandCenterResponse {
  total_receivable: number;
  total_overdue: number;
  total_collected: number;
  total_expected_unbilled: number;
  aging_buckets: CommissionAgingBucketSummary[];
  stage_breakdown: Record<string, { count: number; amount: number }>;
  commissions: CommissionItem[];
}

export interface CommissionStageUpdatePayload {
  stage: CommissionStage;
  invoice_number?: string;
  invoice_date?: string;
  due_date?: string;
  paid_date?: string;
  payment_reference?: string;
  remarks?: string;
}

File: backend/app/schemas/commission_ledger.py
from datetime import datetime
from pydantic import BaseModel
from typing import List, Optional
from app.models.commission import PayoutStatus, CommissionStage

class CommissionStageUpdate(BaseModel):
    stage: CommissionStage
    invoice_number: Optional[str] = None
    invoice_date: Optional[datetime] = None
    due_date: Optional[datetime] = None
    paid_date: Optional[datetime] = None
    payment_reference: Optional[str] = None
    remarks: Optional[str] = None

class CommissionItemResponse(BaseModel):
    id: int
    organization_id: Optional[int] = None
    booking_id: int
    booking_number: str
    unit_number: str
    lead_name: str
    builder_name: str
    project_name: str
    executive_name: str
    broker_name: Optional[str] = None
    total_deal_value: float

    builder_commission_rate: float
    builder_commission_amount: float
    gst_rate: float
    gst_amount: float
    tds_rate: float
    tds_amount: float
    net_receivable: float

    executive_commission_rate: float
    executive_commission_amount: float
    company_margin_amount: float

    stage: str
    payout_status: str
    invoice_number: Optional[str] = None
    invoice_date: Optional[datetime] = None
    due_date: Optional[datetime] = None
    paid_date: Optional[datetime] = None
    payment_reference: Optional[str] = None

    aging_bucket: str
    days_overdue: int
    remarks: Optional[str] = None
    created_at: datetime

    class Config:
        from_attributes = True

class CommissionAgingBucketSummary(BaseModel):
    bucket: str
    count: int
    total_amount: float

class CommissionCommandCenterResponse(BaseModel):
    total_receivable: float
    total_overdue: float
    total_collected: float
    total_expected_unbilled: float
    aging_buckets: List[CommissionAgingBucketSummary]
    stage_breakdown: dict
    commissions: List[CommissionItemResponse]
